frontend/src/api/RestaurantApi.tsx

```diff
@@ -0,0 +1,43 @@
+import { SearchState } from '@/pages/SearchPage';
+import { RestaurantResponse } from '@/types';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useQuery } from 'react-query';
+
+const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
+
+export const useSearchRestaurants = (
+  searchState: SearchState,
+  city: string
+) => {
+  const { getAccessTokenSilently } = useAuth0();
+
+  const createSearchRestaurant = async (): Promise<RestaurantResponse> => {
+    const accessToken = await getAccessTokenSilently();
+    const params = new URLSearchParams();
+    params.set('searchQuery', searchState.searchQuery);
+
+    const response = await fetch(
+      `${API_BASE_URL}/api/restaurant/search/${encodeURIComponent(city)}?${params.toString()}`,
+      {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }
+    );
+
+    if (!response.ok) {
+      throw new Error('Failed to search restaurants');
+    }
+
+    return response.json();
+  };
+
+  const { data: results, isLoading } = useQuery(
+    ['searchRestaurants', searchState],
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/43#issuecomment-2408154714>#86982f</a><span> - 10/11/2024, 09:46 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The query key used in the useQuery hook may not be specific enough, potentially leading to cache issues or unnecessary refetches.
</div>
<div id="fix">
<b>Fix:</b> Update the query key to include all variables that affect the query result. In this case, include both the searchState and city in the query key.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -36,7 +36,7 @@
    };
 
    const { data: results, isLoading } = useQuery(
 -    ['searchRestaurants', searchState],
 +    ['searchRestaurants', searchState, city],
      createSearchRestaurant,
      { enabled: !!city }
    );
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged